pkg/osvscanner/optional_enricher.go

```diff
@@ -0,0 +1,41 @@
+package osvscanner
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// Get the go version of the default environment that osv-scanner is being ran on
+func getGoVersion() (string, error) {
+	versionBytes, err := exec.Command("go", "env", "GOVERSION").Output()
+	if err != nil {
+		return "", err
+	}
+	// version format can be:
+	// - go1.20.6
+	// - go1.22-20230729-RC00 cl/552016856 +457721cd52 X:fieldtrack,boringcrypto
+	version := strings.TrimPrefix(string(versionBytes), "go")
+	version, _, _ = strings.Cut(version, " ")
+
+	return version, nil
+}
+
+func postLockfileEnricher(r reporter.Reporter, parsedLockfile *lockfile.Lockfile) {
```

Comment:
Rather than naming this a generic enricher, let's just call it what it is. e.g. "addCompilerVersion" or something. We can always rename this later if we need to generalize this since it's private.